rust/src/enip/logger.rs

```diff
@@ -0,0 +1,1901 @@
+/* Copyright (C) 2023 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use super::constant::{EnipCommand, EnipStatus};
+use super::enip::EnipTransaction;
+use super::parser::{
+    cip_segment_type_string, CipData, CipDir, EnipCIP, EnipCipPathSegment, EnipCipRequestPayload,
+    EnipCipResponsePayload, EnipHeader, EnipItemPayload, EnipPayload,
+};
+use crate::detect::EnumString;
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+use std;
+
+fn log_enip_header(h: &EnipHeader, js: &mut JsonBuilder) -> Result<(), JsonError> {
+    if let Some(cmd) = EnipCommand::from_u(h.cmd) {
+        js.set_string("command", cmd.to_str())?;
+    } else {
+        js.set_string("command", &format!("unknown-{}", h.cmd))?;
+    }
+    if let Some(status) = EnipStatus::from_u(h.status) {
+        js.set_string("status", status.to_str())?;
+    } else {
+        js.set_string("status", &format!("unknown-{}", h.status))?;
+    }
+    if h.options != 0 {
+        js.set_uint("options", h.options.into())?;
+    }
+    Ok(())
+}
+
+// These lists come from other open source programs like Wireshark, Zeek or scapy
+fn enip_vendorid_string(p: u16) -> Option<&'static str> {
+    match p {
+        1 => Some("Rockwell Automation/Allen-Bradley"),
```

Comment:
I think it's time for us to add a facility to load this kind of data from actual data files, instead of putting it in code. This would lead to cleaner code and it has as additional benefit that data files can be updated independently too.

Reply:
https://redmine.openinfosecfoundation.org/issues/7052